nixos/modules/image/repart-image.nix

```diff
@@ -79,10 +74,11 @@ let
   }."${compression.algorithm}";
 in
 
-runCommand imageFileBasename
-{
+stdenvNoCC.mkDerivation (finalAttrs: {
   __structuredAttrs = true;
 
+  name = imageFileBasename;
```

Comment:
Another nice improvement would be adding a separate `version` attribute.

Reply:
Are you suggesting to use `system.image.id` + `system.image.version` as `pname` and `version` respectively? How should `version == null` be handled?

Reply:
Hmm, going with `pname` + `version = ""` isn't a good idea, that results in `"pname-"` (trailing hyphen) :/
I don't see how this is going to work while still allowing `version` being set to `null` in the module

Reply:
You could add it to the attrset via a conditional merge  `// lib.optionalAttrs (version != null) { inherit version; }`.

Reply:
That would only make sense if I also put in `name` instead of `pname` when `version` is set to null. What would introducing this achieve?

Reply:
It would be very beneficial to have a version attribute available to identify the derivation (e.g. in an SBOM). I understand that it's ugly but I believe its a tradeoff worth taking.

Reply:
Ok, good point! The version is now set together with `pname` if `version` is not null, otherwise only `name` is set.